src/Message/Checkout/PurchaseRequest.php

```diff
@@ -0,0 +1,166 @@
+<?php
+
+/**
+ * Stripe Checkout Session Request.
+ */
+
+namespace Omnipay\Stripe\Message\Checkout;
+
+/**
+ * Stripe Checkout Session Request
+ *
+ * @see \Omnipay\Stripe\Gateway
+ * @link https://stripe.com/docs/api/checkout/sessions
+ */
+class PurchaseRequest extends AbstractRequest
+{
+    /**
+     * Set the success url
+     *
+     * @param string $value
+     *
+     * @return \Omnipay\Common\Message\AbstractRequest|PurchaseRequest
+     */
+    public function setSuccessUrl($value)
+    {
+        return $this->setParameter('success_url', $value);
+    }
+
+    /**
+     * Get the success url
+     *
+     * @return string
+     */
+    public function getSuccessUrl()
+    {
+        return $this->getParameter('success_url');
+    }
+    /**
+     * Set the cancel url
+     *
+     * @param string $value
+     *
+     * @return \Omnipay\Common\Message\AbstractRequest|PurchaseRequest
+     */
+    public function setCancelUrl($value)
+    {
+        return $this->setParameter('cancel_url', $value);
+    }
+
+    /**
+     * Get the success url
+     *
+     * @return string
+     */
+    public function getCancelUrl()
+    {
+        return $this->getParameter('success_url');
```

Comment:
Hi @yepzy 

I'm testing this out and it looks good, I hope the PR gets merged because this is a useful gateway to add. The only bug I've seen so-far is that the getCancelUrl() method accidentally returns the success_url.

```suggestion
        return $this->getParameter('cancel_url');
```